clang-16.yaml

```diff
@@ -0,0 +1,223 @@
+
+  - runs: |
+      cmake --build build
+
+  - runs: |
+      help2man --no-info \
+        --source "Wolfi" \
+        --name "Wolfi Clang ${{package.version}}-r${{package.epoch}}" \
+        --version-string "${{package.version}}-r${{package.epoch}}" \
+        --help-option "--help-hidden" \
+        ./build/bin/clang > clang.1
+
+  - runs: |
+      DESTDIR="${{targets.destdir}}" cmake --install build
+
+      install -Dm644 build/lib/libclang.a -t "${{targets.destdir}}"/usr/lib/
+
+  - runs: |
+      rm "${{targets.destdir}}"/usr/share/clang/clang-format-bbedit.applescript \
+         "${{targets.destdir}}"/usr/share/clang/clang-doc-default-stylesheet.css \
+         "${{targets.destdir}}"/usr/share/clang/index.js
+
+      install -Dm644 clang.1 -t "${{targets.destdir}}"/usr/share/man/man1/
+
+      mkdir -p "${{targets.destdir}}"/usr/share/bash-completion/completions
+      mv "${{targets.destdir}}"/usr//share/clang/bash-autocomplete.sh \
+         "${{targets.destdir}}"/usr/share/bash-completion/completions/clang
+
+      mkdir -p "${{targets.destdir}}"/usr/share/emacs/site-lisp
+      mv "${{targets.destdir}}"/usr/share/clang/clang-*.el \
+         "${{targets.destdir}}"/usr/share/emacs/site-lisp
+
+  - uses: strip
+
+subpackages:
+  - name: "clang-16-dev"
+    description: "headers for clang-16"
+    pipeline:
+      - uses: split/dev
+    dependencies:
+      runtime:
+        - libLLVM-16
+        - libclang-cpp-16
+
+  - name: "libclang-cpp-16"
+    description: "Clang 16 runtime library"
+    pipeline:
+      - runs: |
+          sonamefull="libclang-cpp.so.16"
+
+          mkdir -p "${{targets.subpkgdir}}"/usr/lib/
+          mv "${{targets.destdir}}"/usr/lib/$sonamefull "${{targets.subpkgdir}}"/usr/lib/
+    dependencies:
+      runtime:
+        - libLLVM-16
+
+  - name: "clang-analyzer"
+    description: "Clang 16 analyzer"
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"/usr/bin
+          mv "${{targets.destdir}}"/usr/bin/scan* "${{targets.subpkgdir}}"/usr/bin/
+          mv "${{targets.destdir}}"/usr/bin/analyze-build* "${{targets.subpkgdir}}"/usr/bin/
+          mv "${{targets.destdir}}"/usr/bin/intercept-build* "${{targets.subpkgdir}}"/usr/bin/
+
+          mkdir -p "${{targets.subpkgdir}}"/usr/libexec
+          mv "${{targets.destdir}}"/usr/libexec/*-analyzer* "${{targets.subpkgdir}}"/usr/libexec/
+          mv "${{targets.destdir}}"/usr/libexec/analyze-* "${{targets.subpkgdir}}"/usr/libexec/
+          mv "${{targets.destdir}}"/usr/libexec/intercept-* "${{targets.subpkgdir}}"/usr/libexec/
+
+          mkdir -p "${{targets.subpkgdir}}"/usr/share
+          mv "${{targets.destdir}}"/usr/share/scan-* "${{targets.subpkgdir}}"/usr/share/
+
+          mkdir -p "${{targets.subpkgdir}}"/usr/lib
+          mv "${{targets.destdir}}"/usr/lib/libear* "${{targets.subpkgdir}}"/usr/lib/
+    dependencies:
+      runtime:
+        - perl
+        - python3
+
+  - name: "clang-extras"
```

Comment:
I think we also need `-16` suffix so we don't clash with https://github.com/wolfi-dev/os/blob/48479a25be862435f1420065cbdc677ae864272b/clang-15.yaml#L160

Reply:
Good catch. We should probably switch this to be slotted in the future.